swarm/src/connection/substream.rs

```diff
@@ -134,7 +134,7 @@ where
     pub fn poll(
         &mut self,
         cx: &mut Context<'_>,
-    ) -> Poll<Result<SubstreamEvent<TMuxer, TUserData>, IoError>> {
+    ) -> Poll<Result<SubstreamEvent<TMuxer, TUserData>, TMuxer::Error>> {
```

Comment:
Not in scope for this PR, but while reviewing I was wondering: Do we even need the `TMuxer` generic on `Muxing`? I think the actual type is always `StreamMuxerBox`. `Muxing` is private the swarm crate and only used by `Connection` and there we already use the concrete `StreamMuxerBox`.

Reply:
I am going to inline the entire `Muxing` type into `Connection` once `poll_outbound` is removed from `StreamMuxer`. This is already done in #2648 if you want to see what it looks like :)

I am also tempted to inline `HandlerWrapper` and centralize all the connection handling logic directly in `Connection`. This would remove quite a bit of indirection because substreams would directly be passed from the `StreamMuxer` to the `ConnectionHandler`.